src/Symfony/Component/Console/Tester/CommandTester.php

```diff
@@ -24,9 +27,12 @@ class CommandTester
 {
     use TesterTrait;
 
+    private Command $command;
+
     public function __construct(
-        private Command $command,
+        callable|Command $command,
     ) {
+        $this->command = is_callable($command) ? new Command($command) : $command;
```

Comment:
No need to create a new `Command` object if `$command` is an invokable class extending the `Command`. Could you add a test for that, similar to `ApplicationTest::testAddCommandWithInvokableExtendedCommand()`.

```suggestion
        $this->command = $command instanceof Command ? $command : new Command(null, $command);
```

Reply:
Good one, added the tests too.